### PR TITLE
[supervisor] enable PWA

### PR DESCRIPTION
**Pull request by akosyakov**
#### What it does

- It allows to start a new workspace as an app to enable desktop like experience.

#### How to test

- Switch to Code
- Start a workspace
- You should see start a PWA icon in the right corner of URl. See https://support.google.com/chrome/answer/9658361?co=GENIE.Platform%3DDesktop&hl=en
- Click it, try to open a file and user Ctrl+W to check whether it closes the editor, not the window.

### CODE REVIEW

File: components/ws-proxy/pkg/proxy/routes.go [hunk 1]
```diff
@@ -75,16 +75,19 @@ func installWorkspaceRoutes(r *mux.Router, config *RouteHandlerConfig, ip Worksp
 	//       Routes registered first have priority over those that come afterwards.
 	routes := newIDERoutes(config, ip)
 
-	// The favicon warants special handling, because we pull that from the supervisor frontend
+	// The supervisor frontend resources warant special handling, because we pull them from the supervisor frontend
```
Comment (akosyakov) [inline components/ws-proxy/pkg/proxy/routes.go:78]: testset

**Review by akosyakov — COMMENTED** [review]
test

**Comment by akosyakov** [comment]
> More generally: are there consequences of this change other than Ctrl+W working?

It will generally behave more like desktop app. You should see all apps in chrome://apps/

**Comment by csweichel** [comment]
I tried with `Cmd+W` on OSX and it closed my tab :/

**Comment by akosyakov** [comment]
> I tried with Cmd+W on OSX and it closed my tab :/

It seems to be an expected behaviour in VS Code if you don't have any editors opened. On Desktop it is also closing the window.

**Comment by csweichel** [comment]
> > I tried with Cmd+W on OSX and it closed my tab :/
> 
> It seems to be an expected behaviour in VS Code if you don't have any editors opened. On Desktop it is also closing the window.

I did have an editor open though

**Comment by akosyakov** [comment]
Another potential issue is working with Clipboard, in the browser each user should grant access for a new workspace and a user get asked only if the clipboard action is triggered in the trusted event, i.e. triggered by a user, not pragmatically. Maybe it could be addressed by PWA as well.

**Comment by akosyakov** [comment]
> Another potential issue is working with Clipboard, in the browser each user should grant access for a new workspace and a user get asked only if the clipboard action is triggered in the trusted event, i.e. triggered by a user, not pragmatically. Maybe it could be addressed by PWA as well.

nope, it won't help, permissions, caches and storages are scoped per origin. It can be only configured by a user to allow for subdomains with wildcard syntax.

**Comment by akosyakov** [comment]
Closed in favor of https://github.com/gitpod-io/gitpod/pull/3699

**Pull request closed without merging**